pandas/core/frame.py

```diff
@@ -1258,6 +1258,11 @@ def from_items(cls, items, columns=None, orient='columns'):
         """
         keys, values = lzip(*items)
 
+        for val in values:
```

Comment:
NO, this is completely non-performant.

you need to catch the error and then do the check.